### ISSUE TITLE
Moderation Manager

### ISSUE THREAD

**Issue by binaryoverload** [CascadeBot/CascadeBot#51]
It will be good for us to have a central moderation manager that allows all moderation actions to have consistency in terms of errors and responses and also allow easy integration with the modlog system when we decide to do it

**Comment by binaryoverload** [comment]
Moderation actions that need to be implemented:
•  BAN 
•  KICK 
•  SOFTBAN 
•  WARN 
•  MUTE

**Comment by binaryoverload** [comment]
The main method for this will be a `handleModeration` method having the following parameters:
 - `context` - The command context to be able to access guild information as well as respond back to the user
 - `action` - The moderation action to be executed, this will be one of the items off [the list](https://github.com/CascadeBot/CascadeBot/issues/51#issuecomment-466790680) 
 - `target` - The `Member` that the action is to be executed against
 - `submitter`  - The `Member` that the action was executed by
 - `reason` - The possible blank/null reason for this action. Whether this reason can be blank is down to the different moderation actions which will respond appropriately

**Comment by binaryoverload** [comment]
@CascadeBot/cascadebot-contributors Anything to add to this?

**Comment by JamesHawkinss** [comment]
Looks good at the moment, this would simplify things a lot and would also make creating new moderation commands.

Would we be able to make this sort of manager for informational commands as well, or would it not be worth it? @BinaryOverload

**Comment by binaryoverload** [comment]
What sort of manager have you got in mind for informational? I don't see a common feature we could be able to have across different informational commands. Maybe a better way to do different embed fields but it wouldn't be a manager.

**Comment by JamesHawkinss** [comment]
Perhaps not for specifically informational but instead one for error messages? We seem to reuse several of (nearly) the same errors and having a centralised system for calling error messages might be good for continutity between commands.

**Comment by binaryoverload** [comment]
Draft up your ideas and examples into another issue, I like where you're coming from.

**Comment by binaryoverload** [comment]
@weeryan17 Anything you would like to see make it into the new feature?

**Comment by weeryan17** [comment]
Not really. the only thing I'd like to mention is if we're going with on central method that method is going to be really complicated. maybe we could have a method for temp and a different method for everything else? Also i noticed you didn't bring up temp punishments here so we should plan those out as well cause i feel like those should be in the moderation manager.

**Comment by binaryoverload** [comment]
Well my plan is to have this one method call different methods so it won't be one huge method. And the system will allow for temp just fine I just won't implement it

**Comment by weeryan17** [comment]
Sounds good then

**Comment by binaryoverload** [comment]
@mrjvs Any insights on how we should do the UX for displaying various messages relating to moderation actions?

**Comment by mrjvs** [comment]
Wasn’t expecting this oof, i don’t have any ideas atm. I’ll get back to this tomorrow as I’m taking a day off to relax.

**Comment by mrjvs** [comment]
https://i.imgur.com/Os34UJn.png here is visual representation of responses. All of the errors can and should be in the same format for all moderation actions (maybe phrase it a bit different, im looking at you binary. it should not be fields tho, sentences should flow easily instead of just stating stuff). format should also be consistent for the command structure, the last thing you want when banning a user is forgetting the format of the command. The colors (which are not final) should probably be put in an enum because those will be consistent throughout the entire bot.

**Comment by binaryoverload** [comment]
@mrjvs Awesome thanks. On the context of colours, at the moment we just use standard red, green and yellow colours provided by Java. Did you want us to continue using these or did you want to have others?

**Comment by mrjvs** [comment]
@binaryoverload custom colors probably, just have an enum with colors ready for easy modification for later.

**Comment by binaryoverload** [comment]
> @binaryoverload custom colors probably, just have an enum with colors ready for easy modification for later.

Sounds good, I have opened #53 relating to this

**Comment by binaryoverload** [comment]
Waiting on #46 so I can integrate the manager into in
